braindecode/samplers/base.py

```diff
@@ -81,6 +87,10 @@ def __iter__(self):
     def n_recordings(self):
         return self.info.shape[0]
 
+    @property
```

Comment:
Might be actually safer to supply n_classes explicitly? Otherwise you may run into problems:
* if in some subset not all classes are present (imagine small evaluation subset in debugging experiment)
* targets somewhere else as in https://github.com/braindecode/braindecode/pull/261 may not work.

Reply:
The first point is fixed in the latest commit. I'm not sure about the second point though - the sampler requires categorical targets in the BaseConcatDataset's `metadata` attribute. What use case exactly did you have in mind @robintibor ?